game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleSteps.java

```diff
@@ -185,20 +189,14 @@ public boolean isAmphibious() {
       steps.addAll(offensiveFirstStrike.getNames());
     }
     steps.addAll(airAttackVsNonSubs.getNames());
+    steps.addAll(offensiveStandard.getNames());
 
-    if (attackingUnits.stream().anyMatch(Matches.unitIsFirstStrike().negate())) {
-      steps.add(attacker.getName() + FIRE);
```

Comment:
Side-note, it seems like this class knows quite a lot:
- how to form step names (EG:  `attacker.getName() + FIRE`
- how to extract step names (EG: `steps.addAll(offensiveFirstStrike.getNames());`)
- the ordering of steps
- if to offer certain steps (evaluates retreat rules, eg: line 215 (if (RetreatChecks.canAttackerRetreat(...)

Any thoughts about this long term and if we're heading in a direction to fix this?

Reply:
Once all of the steps are converted, it will just create the steps, then order them, and then map them to their names.

See `MustFightBattle#addFightStartSteps` at https://github.com/triplea-game/triplea/blob/1cd72b85200ec7e9de2890aad83a034332b1ffb4/game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java#L1051 for a rough example of what it would look like.